source/features/pull-request-timeline-filter.tsx

```diff
@@ -0,0 +1,260 @@
+
+function restoreSettings(): void {
+	(select('#' + hideUnresolvedSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideUnresolved;
+	(select('#' + hideResolvedSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideResolved;
+	(select('#' + hideCommitsSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideCommits;
+	(select('#' + hideNormalCommentsSelectorId) as HTMLInputElement)!.checked = CurrentSettings.hideNormalComment;
+	(select('#' + hideOthersSelectorId) as HTMLInputElement)!.checked = CurrentSettings.HideOthers;
+	(select('#' + autoLoadHiddenSelectorId) as HTMLInputElement)!.checked = CurrentSettings.AutoLoadHidden;
+}
+
+function createItem(form: JSX.Element, id: string, title: string, summary: string, isSelected: boolean, hasTopBorder: boolean): void {
+	const element = (
+		<label className={'d-block p-3 ' + (hasTopBorder ? 'border-top' : '')}>
+			<div className="form-checkbox my-0">
+				<input id={id} type="checkbox" name="id" value="unsubscribe" checked={isSelected}/> {title}
+				<p className="note">
+					{summary}
+				</p>
+			</div>
+		</label>
+	);
+
+	form.append(element);
+}
+
+async function addTimelineItemsFilter(): Promise<void> {
+	const notifications = await elementReady(notiticationsSelector);
+	if (!notifications) {
+		return;
+	}
+
+	// Copy existing element and adapt its content
+	const timelineFilter = notifications.cloneNode(true);
+	timelineFilter.id = 'timeline-filters';
+
+	select('form.thread-subscribe-form', timelineFilter)!.remove();
+	const summary = select('summary', timelineFilter)!;
+	summary.setAttribute('aria-label', 'Customize timeline filters');
+	select('div.text-bold', summary)!.textContent = 'Filters';
+
+	createDetailsDialog(timelineFilter);
+	notifications.after(timelineFilter);
+
+	regenerateFilterSummary();
+}
+
+function createDetailsDialog(timelineFilter: Element): void {
+	const detailsDialog = select('details-dialog', timelineFilter)!;
+
+	detailsDialog.setAttribute('src', '');
+
+	detailsDialog.setAttribute('aria-label', 'Timeline filter settings');
+	select('div.Box-header h3', detailsDialog)!.textContent = 'Timeline filter settings';
+
+	// Close button should restore previous settings.
+	select('div.Box-header button', detailsDialog)!.addEventListener('click', restoreSettings);
+
+	const form = <div/>;
+
+	createItem(form, hideResolvedSelectorId, 'Hide resolved comments', '', CurrentSettings.HideResolved, false);
+	createItem(form, hideCommitsSelectorId, 'Hide commits', '', CurrentSettings.HideCommits, true);
+	createItem(form, hideUnresolvedSelectorId, 'Hide unresolved comments', '', CurrentSettings.HideUnresolved, true);
+	createItem(form, hideNormalCommentsSelectorId, 'Hide normal comments', 'Hides comments that does not contain unresolved/resolved state.', CurrentSettings.hideNormalComment, true);
+	createItem(form, hideOthersSelectorId, 'Hide other', 'Hides any other kind of activity that was not specified above.', CurrentSettings.HideOthers, true);
```

Comment:
Instead of passing `form`, you could build the whole form in a single JSX by returning the generated element:

``` JSX
const form = (
  <form>
    {createItem(etc)}
    {createItem(etc)}
  
   </form>
)
```

Reply:
Done